tests/js/set-tests.js

```diff
@@ -0,0 +1,397 @@
+            }
+        };
+
+        const realm = new Realm({ schema: [people_schema] });
+        const schema = realm.schema;
+
+        realm.write(() => {
+            // insert two people
+            realm.create(people_schema.name, {
+                Names: ["John", "Sue"]
+            });
+        });
+
+        let people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(2, people[0].Names.length, "Persons Set length should be 2");
+
+        // add another three names
+        realm.write(() => {
+            people[0].Names.add("Dev").add("Ava").add("Harsha");
+        });
+        people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(5, people[0].Names.length, "Persons Set length should be 5");
+
+        // remove two names
+        realm.write(() => {
+            people[0].Names.delete("John");
+            people[0].Names.delete("Sue");
+        });
+        people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(3, people[0].Names.length, "Persons Set length should be 3");
+        
+        realm.close();
+    },
+
+
+    async testSetObjectAddDelete() {
+        const person_schema = {
+            name: "Person",
+            properties: {
+                FirstName: 'string',
+                Age: 'int'
+            }
+        };
+
+        const people_schema = {
+            name: 'People',
+            properties: {
+                Persons: 'Person<>'
+            }
+        };
+
+        const realm = new Realm({ schema: [person_schema, people_schema] });
+        const schema = realm.schema;
+
+        realm.write(() => {
+            // insert two people
+            realm.create(people_schema.name, {Persons: [
+                {FirstName: "Joe",  Age: 4},
+                {FirstName: "Sue",  Age: 53},
+            ]});
+        });
+
+
+        let people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(2, people[0].Persons.length, "Persons Set length should be 2");
+
+        // add another person
+        realm.write(() => {
+            people[0].Persons.add({FirstName: 'bob', Age: 99});
+        });
+
+        people = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, people.length, "There should be one 'People' entry")
+        TestCase.assertEqual(3, people[0].Persons.length, "Persons Set length should be 3");
+//        console.log("Joe's name:  " + people.Persons[0].FirstName);
+        let onfe = people[0].Persons.get(0);
+
+        let filteredsues = people[0].Persons.filtered('FirstName = "Sue"');
+
+        console.log("Sue's age:  " + filteredsues[0].Age);
+
```

Comment:
There doesn't seem to be any assertions on `filteredsues`.  Is this test perhaps not complete?

Reply:
Yeah, still working on more coverage on unit tests.  I'll make sure to incorporate the feedback, though :+1: